doc/source/train/examples/lightning/lightning-llm-finetuning-3b.ipynb

```diff
@@ -0,0 +1,888 @@
+    "scaling_config = ScalingConfig(\n",
+    "    num_workers=num_workers, use_gpu=True, resources_per_worker={\"CPU\": 12, \"GPU\": 1}\n",
+    ")\n",
+    "\n",
+    "trainer = LightningTrainer(\n",
+    "    lightning_config=lightning_config.build(),\n",
+    "    run_config=run_config,\n",
+    "    scaling_config=scaling_config,\n",
+    "    datasets={\"train\": ray_datasets[\"train\"]},\n",
+    "    datasets_iter_config={\"batch_size\": batch_size_per_worker},\n",
+    "    preprocessor=preprocessor,\n",
+    ")\n",
+    "result = trainer.fit()\n",
+    "\n",
+    "result\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We finished training in 553s(9min 13s). One on-demand g4dn.4xlarge cost `$1.204/h`. The total cost would be `$1.204 * 16 * 553 / 3600 = $2.96`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "download: s3://anyscale-staging-data-cld-kvedzwag2qa8i5bjxuevf5i7/ray-lightning-results/finetune_dolly-v2-3b/LightningTrainer_f8b8d_00000_0_2023-05-03_00-49-02/checkpoint_000000/.is_checkpoint to ../../../../../../../../tmp/model-checkpoint/.is_checkpoint\n",
+      "download: s3://anyscale-staging-data-cld-kvedzwag2qa8i5bjxuevf5i7/ray-lightning-results/finetune_dolly-v2-3b/LightningTrainer_f8b8d_00000_0_2023-05-03_00-49-02/checkpoint_000000/_preprocessor to ../../../../../../../../tmp/model-checkpoint/_preprocessor\n",
+      "download: s3://anyscale-staging-data-cld-kvedzwag2qa8i5bjxuevf5i7/ray-lightning-results/finetune_dolly-v2-3b/LightningTrainer_f8b8d_00000_0_2023-05-03_00-49-02/checkpoint_000000/.tune_metadata to ../../../../../../../../tmp/model-checkpoint/.tune_metadata\n",
+      "download: s3://anyscale-staging-data-cld-kvedzwag2qa8i5bjxuevf5i7/ray-lightning-results/finetune_dolly-v2-3b/LightningTrainer_f8b8d_00000_0_2023-05-03_00-49-02/checkpoint_000000/.metadata.pkl to ../../../../../../../../tmp/model-checkpoint/.metadata.pkl\n",
+      "download: s3://anyscale-staging-data-cld-kvedzwag2qa8i5bjxuevf5i7/ray-lightning-results/finetune_dolly-v2-3b/LightningTrainer_f8b8d_00000_0_2023-05-03_00-49-02/checkpoint_000000/model to ../../../../../../../../tmp/model-checkpoint/model\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "0"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "checkpoint_uri = result.checkpoint.uri\n",
+    "checkpoint_local_dir = \"/tmp/model-checkpoint\"\n",
+    "# checkpoint_uri = \"s3://yunxuanx-test/model-checkpoint/finetune-dolly-v2/LightningTrainer_18e2e_00000_0_2023-04-30_17-57-08/checkpoint_000000/\"\n",
+    "\n",
+    "os.makedirs(checkpoint_local_dir, exist_ok=True)\n",
+    "os.system(f\"aws s3 sync {checkpoint_uri} {checkpoint_local_dir}\")"
```

Comment:
I see. potentially this can be done via a Checkpoint class.
do these FSDP model run differently from normal PyTorch models?

Reply:
I've tried `Checkpoint.from_uri()`, but the downloading speed is extremely slow. 

Actually the checkpoint has state dict for unsharded model (fsdp wrapper prefix get trimmed). We can re-shard it later by ourselves.